File: Sorting/Selection Sort/C#/SelectionSort.cs
public class SelectionSort {
    public static void selectionSort(int[] array) 
    {
        for (int i = 0; i < array.Length - 1; i++)
        {
            int min_index = i;
            for (int j = i + 1; j < array.Length; j++)
                if (array[j] < array[min_index])
                    min_index = j;    //Initial minimum index is updated
 
            
            int temp = array[min_index]; //Then it swapped the smallest element with current index element
            array[min_index] = array[i];
            array[i] = temp;          
        }
    }

    //MAIN METHOD
    /*
    public static void Main()
    {
        int[] a = { 23, 25, 22, 21, 27 };
        selectionSort(a);

        //After sorted
        for (int i = 0; i < a.Length; i++)
        {
            if(i != a.Length-1)
                Console.Write(a[i] + ", ");
            else
                Console.Write(a[i]);
        }
    }
    */


    /*
    Time complexity in the worst case: O(N^2)
    Space complexity: O(1)
    */
}

File: Sorting/Selection Sort/Java/SelectionSort.java
public class SelectionSort {	
	public static int[] selectionSort(int[] array) {
		if (array == null || array.length == 0) //if array is not sortable, then return null
			return null;		
		for (int i = 0; i < array.length-1; i++) {
			int minIndex = i;
			for (int j = i + 1; j < array.length;j++) //we find the index of smallest element at the rest of the array in this for loop
				if (array[j] < array[minIndex])
					minIndex = j;
			int temp = array[i]; //we swap the smallest element with current index element
			array[i] = array[minIndex];
			array[minIndex] = temp;
		}		
		return array;
	}
}
	
	/*
	 * Sample Input: [4,3,2,1]
	 * Sample Output: [1,2,3,4]
	 * 
	 * Time Complexity in Best Case: O(n^2)
	 * Time Complexity in Average Case: O(n^2)
	 * Time Complexity in Worst Case: O(n^2)
	 * Space Complexity in Worst Case: O(1) 
	 */